PWGHF/TableProducer/treeCreatorToXiPi.cxx

```diff
@@ -226,15 +226,15 @@ struct HfTreeCreatorToXiPi {
     rowEv(collision.sel8(), std::abs(collision.posZ()) < cutZPv);
   }
 
-  template <class TMyTracks, typename T>
+  template <class TMyTracks, class TMyEvents, typename T>
   void fillCandidate(const T& candidate, int8_t flagMc, int8_t debugMc, int8_t originMc, bool collisionMatched)
   {
     rowCandidateFull(
       candidate.xPv(),
       candidate.yPv(),
       candidate.zPv(),
-      candidate.collision().numContrib(),
-      candidate.collision().chi2(),
+      candidate.template collision_as<TMyEvents>().numContrib(),
+      candidate.template collision_as<TMyEvents>().chi2(),
```

Comment:
Hi @fgrosa, I used the template following the suggestion Fabio gave me for the variables https://github.com/AliceO2Group/O2Physics/blob/247837f66f0f9a439a5ec707c8965349c87ba7bc/PWGHF/TableProducer/treeCreatorToXiPi.cxx#L315-L316 (this was the PR https://github.com/AliceO2Group/O2Physics/pull/4672,  I thought the two cases were equivalent)

Reply:
In the PR that you linked me there is no such implementation, i.e. no additional template parameter is added (you indeed use directly `aod::TrackSelection`), so maybe you were referring to another PR? 
I am asking because here in principle there is no need of a template parameter, since you have always one type of tracks (but in any case it should probably be `candidate.template collision_as<MyEventTable>()`, since `candidate` is instead a template parameter)

Reply:
I thought the point was how to do the dereferencing of the index in a template function. Yes, in that PR I used directly TrackSelection, then I changed in this PR https://github.com/AliceO2Group/O2Physics/pull/4709/files